bot.py

```diff
@@ -32,153 +29,59 @@
-
-
-NEW_YEAR_PATTERNS = load_patterns_from_files(NEW_YEAR_PATTERNS_FILE)
-
-
-async def show_selected_users():
-    async for dialog in CLIENT.iter_dialogs():
-        if dialog.id in CIRCULATION_IDS:
-            logging.info(f"Selected username: {dialog.name}; ID: {dialog.id}")
-
-
-async def filter_f(event) -> bool:
-    for word in NEW_YEAR_PATTERNS:
-        if word in str(event.raw_text).lower():
-            return True
+    current_date = datetime.today().strftime("%m.%d")
+    if db.get_tg_id(current_date) is None:
+        logging.info(f"Today - {current_date} - are no one to congratulate")
+        time.sleep(86400)
     else:
-        return False
-
-
-@CLIENT.on(events.NewMessage(from_users=CIRCULATION_IDS, func=filter_f))
-async def reply_to_congratulations(event):
-    user_data = await event.client.get_entity(event.from_id)
-    logging.info(
-        f"Contact: {user_data.contact} - "
-        f"first name: {user_data.first_name} - "
-        f"ID: {user_data.id} - "
-        f"sent congratulation: {event.message.message}"
-    )
-    async with CLIENT.action(user_data.id, "typing"):
-        await asyncio.sleep(random.randrange(5, 10))
-        await event.reply(event.message.message)
-
-
-async def send_message_template(
-    user_data, event, start_range, end_range, response_type
-):
-    logging.info(
-        f"Contact: {user_data.contact} - "
-        f"first name: {user_data.first_name} - "
-        f"ID: {user_data.id} - "
-        f"sent message: {event.message.message}"
-    )
-
-    logging.info("Waiting for response...")
-
-
-# TODO send always
-#     async with CLIENT.action(user_data.id, "typing"):
-#         await asyncio.sleep(random.randrange(start_range, end_range))
-#         await CLIENT.send_message(
-#             user_data.id,
-#             f"""
-# Hello, {user_data.first_name}. \n
-# **This message was sent automatically.** \n
-# """,
-#         )
-#         await CLIENT.send_message(user_data.id, response_type)
-#         logging.info(f"Response was sent to {user_data.first_name}.")
-
-
-@CLIENT.on(events.NewMessage(incoming=True, from_users=CIRCULATION_IDS))
-async def response_to_group(event):
-    await show_selected_users()
-
-    user_data = await event.client.get_entity(event.from_id)
+        user_data = await CLIENT.get_entity(db.get_tg_id(current_date))
+        logging.info(
+            f"Today - {current_date} - going to congratulate {user_data.first_name} - {user_data.username}"
+        )
+        await CLIENT.send_message(
+            db.get_tg_id(current_date), random.choice(CONGRATULATIONS_FILE)
+        )
+        async with CLIENT.action(user_data.id, "typing"):
+            await asyncio.sleep(random.randrange(2, 5))
+            await CLIENT.send_file(
+                db.get_tg_id(current_date),
+                bucket.download_file_from_bucket(Path("/home/")),
+            )
+    time.sleep(400)
+    bucket.prune_directory(Path("/home/"))
+    time.sleep(86000)
```

Comment:
Wow... There might be an issue in your architecture.

At least, put a comment how many minutes/hours it is. Also, you can write `86_000` do make numbers be more distinct.

Reply:
It is calculated for sure. But it is really awfull solution. In fututre I am going to use cloud solutions to run the code by cron. Someday